nautobot/docs/development/release-checklist.md

```diff
@@ -1,46 +1,66 @@
+This would allow Poetry to install `djangorestframework` versions `>=3.12.2` but `<3.13.0`.
+
+#### The `poetry.lock` file
+
+The other file is `poetry.lock`, which is managed by Poetry and contains package names, versions, and other metadata.
+
+Each of the required packages pinned to its current stable version. When Nautobot is installed, this file is used to resolve and install all dependencies listed in `pyproject.toml`, but Poetry will use the exact versions found in `poetry.lock` to ensure that a new release of a dependency doesn't break Nautobot.
+
+!!! warning
+    You must never directly edit this file. You will use `poetry update` commands to manage it.
+
+#### Run `poetry update`
+
+Every minor version release should refresh `poetry.lock`, so that it lists the most recent stable release of each package. To do this:
 
-Every minor version release should refresh `requirements.txt` so that it lists the most recent stable release of each package. To do this:
+1. Run `poetry install` to install the latest version of all required packages.
+2. Run all tests and check that the UI and API function as expected.
+3. Review each requirement's release notes for any breaking or otherwise noteworthy changes.
+4. If a required package requires updating to a new release not covered in the version constraints for a package as defined in `pyproject.toml`, (e.g. `Django ~3.1.7` would never install `Django >=4.0.0`), update it manually in `pyproject.toml`.
+5. Run `poetry update <package>` to update the package versions in `poetry.lock` as appropriate.
 
-1. Create a new virtual environment.
-2. Install the latest version of all required packages `pip install -U -r base_requirements.txt`).
-3. Run all tests and check that the UI and API function as expected.
-4. Review each requirement's release notes for any breaking or otherwise noteworthy changes.
-5. Update the package versions in `requirements.txt` as appropriate.
+!!! hint
+    You may use `poetry update --dry-run` to have Poetry automatically tell you what package updates are avaiable and the versions it would upgrade.
 
 ### Update Static Libraries
 
 Update the following static libraries to their most recent stable release:
 
-* Bootstrap 3
-* Material Design Icons
-* Select2
-* jQuery
-* jQuery UI
+* [Bootstrap 3](https://getbootstrap.com/docs/3.4)
+* [Material Design Icons](https://materialdesignicons.com/)
+* [Select2](https://github.com/select2/select2/releases)
+* [jQuery](https://jquery.com/download/)
+* [jQuery UI](https://jqueryui.com/)
 
 ### Link to the Release Notes Page
 
 Add the release notes (`/docs/release-notes/X.Y.md`) to the table of contents within `mkdocs.yml`, and point `index.md` to the new file.
 
 ### Manually Perform a New Install
```

Comment:
This section heading is no longer accurate as we're no longer installing Nautobot as part of this section. Do we want to add some content about working through the entire install process (as documented elsewhere in the docs) to make sure nothing there needs to be changed or updated?

Reply:
I don't know. I'm of the mind that this process could be automated so long as we continue to assert that the manual install steps are the reference script from which automated deployments are derived?

Reply:
I agree that deployment testing *should* be part of the automated CI/CD for any push to `main`. Until we have that in place, we probably should continue to do manual installs as part of the pre-release validation.